griptape/artifacts/base_artifact.py

```diff
@@ -1,15 +1,20 @@
 from __future__ import annotations
 from griptape.mixins import SerializableMixin
-from typing import Any
+from typing import Any, TYPE_CHECKING, Optional
 import json
 import uuid
 from abc import ABC, abstractmethod
 from attrs import define, field, Factory
 
+if TYPE_CHECKING:
+    from griptape.common import Reference
 
-@define()
+
+@define
 class BaseArtifact(SerializableMixin, ABC):
     id: str = field(default=Factory(lambda: uuid.uuid4().hex), kw_only=True, metadata={"serializable": True})
+    reference: Optional[Reference] = field(default=None, kw_only=True, metadata={"serializable": True})
```

Comment:
`Reference`s are very specific to a particular RAG Module, does this belong on all Artifacts? What if we created a `ReferenceArtifact` with `Reference` as an inner class?

Reply:
I want reference to be uniformly adopted throughout the framework where it makes sense. For example, I want all loaders to support it.